Client/Frontend/Browser/FaviconHandler.swift

```diff
@@ -25,82 +25,23 @@ class FaviconHandler {
 
   @MainActor func loadFaviconURL(
     _ faviconURL: String,
-    type: IconType,
     forTab tab: Tab
-  ) async throws -> (Favicon, Data?) {
-    guard let iconURL = URL(string: faviconURL), let currentURL = tab.url else {
+  ) async throws -> Favicon {
+    guard let currentURL = tab.url else {
       throw FaviconError.noImageLoaded
     }
     
-    var imageOperation: SDWebImageOperation?
-
-    let webImageCache = WebImageCacheManager.shared
-    
-    return await withCheckedContinuation { continuation in
-      let onProgress: ImageCacheProgress = { receivedSize, expectedSize, _ in
-        if receivedSize >= FaviconHandler.maximumFaviconSize || expectedSize > FaviconHandler.maximumFaviconSize {
-          imageOperation?.cancel()
-        }
-      }
-      
-      let onSuccess: (Favicon, Data?) -> Void = { [weak tab] (favicon, data) -> Void in
-        defer { continuation.resume(returning: (favicon, data)) }
-        
+    return try await withCheckedThrowingContinuation { continuation in
+      FaviconFetcher.loadIcon(url: currentURL, kind: .smallIcon, persistent: !tab.isPrivate) { [weak tab] favicon in
         guard let tab = tab else { return }
         
-        tab.favicons.append(favicon)
-        FaviconMO.add(favicon, forSiteUrl: currentURL, persistent: !tab.isPrivate)
-      }
-      
-      let onCompletedSiteFavicon: ImageCacheCompletion = { image, data, _, _, url in
-        let favicon = Favicon(url: url.absoluteString, date: Date(), type: type)
-        
-        guard let image = image,
-              let imageData = data
-        else {
-          favicon.width = 0
-          favicon.height = 0
-          
-          onSuccess(favicon, data)
-          return
+        if let favicon = favicon {
+          tab.favicons.append(favicon)
```

Comment:
This shouldn't occur in the continuation like this, you should get the result and add it after like:

```swift
let favicon = try await withCheckedThrowingContinuation { ... }
tab.favicons.append(favicon)
return favicon
```